lib/AST/ArchetypeBuilder.cpp

```diff
@@ -2058,12 +2074,26 @@ void ArchetypeBuilder::dump(llvm::raw_ostream &out) {
   out << "\n";
 }
 
-void ArchetypeBuilder::addGenericSignature(GenericSignature *sig) {
+void ArchetypeBuilder::addGenericSignature(GenericSignature *sig,
+                                           bool protocolRequirementSig) {
   if (!sig) return;
-  
-  RequirementSource::Kind sourceKind = RequirementSource::Explicit;
-  for (auto param : sig->getGenericParams())
+
+  if (protocolRequirementSig) {
+    assert(sig->getGenericParams().size() == 1 &&
+           sig->getRequirements().size() == 1 &&
+           sig->getRequirements()[0].getKind() ==
+               RequirementKind::Conformance &&
+           "computing protocol requirement signature with invalid signature");
+  }
```

Comment:
This is the bit that convinced me that we don't want the `protocolRequirementSig` parameter.